noxfile.py

```diff
@@ -180,3 +181,38 @@ def make(session: nox.Session):
         *(args.make_args or ("clean", "coredocs")),
     ]
     session.run("make", "-C", "docs/docsite", *make_args, external=True)
+
+
+def _rm_tmpdir(tmpdir):
+    rmtree(tmpdir, ignore_errors=True)
+
+
+@nox.session
+def tag(session: nox.Session):
+    """
+    Determine the missing ansible-core releases,
+    create corresponding tags for each release in the ansible-documentation repo,
+    and then push them
+    """
+    install(session, req="tag")
+
+    DEFAULT_REPO = "https://github.com/ansible/ansible"
+    CHECK_LOCATION = Path.cwd().parents[0] / "ansible"
+
+    args = list(session.posargs)
+    if not any(arg.startswith("--core") for arg in args):
+        if (CHECK_LOCATION / ".git").is_dir():
+            session.run("git", "-C", CHECK_LOCATION, "pull", "--quiet", external=True)
+            args.extend(["--core", str(CHECK_LOCATION)])
+        else:
+            tmpdir = session.create_tmp()
+            session.run("git", "clone", "--quiet", DEFAULT_REPO, tmpdir, external=True)
+            args.extend(["--core", tmpdir])
+
+    # If run without any commands, default to "tag"
+    if not any(arg.startswith(("hash", "mantag", "new-tags", "tag")) for arg in args):
+        args.append("tag")
```

Comment:
Another thing to consider here is the `ansible-documentation` repo. So far the session can grab the core repo and handle the running of the tag script. But the tagger evaluates the docs repo against the core repo: https://github.com/ansible/ansible-documentation/blob/3551f1999bb18fdfcd5ff2cbf85c33f42000be21/hacking/tagger/tag.py#L206

Maybe we should clone of both repos into the virtual environment?